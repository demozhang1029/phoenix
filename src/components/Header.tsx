import React from "react";
import { View, Text, StyleSheet } from "react-native";
import { white } from "src/config/color";

interface IHeader {
  title: string;
}

export class Header extends React.Component<IHeader> {
  public render() {
    const {title} = this.props;
    return (
      <View style={styles.container}>
        <Text style={styles.title}>{title}</Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    height: "100%",
    display: "flex",
    flexDirection: "row",
    alignItems: "center",
  },
  title: {
    fontSize: 30,
    color: white,
  }
});
