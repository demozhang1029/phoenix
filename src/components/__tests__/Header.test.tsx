import React from 'react';
import { render } from '@testing-library/react-native';
import { Header } from '../Header';

describe('Header', () => {
  test('should render header correctly', () => {
    const { baseElement } = render(<Header title={'hah'} />);

    expect(baseElement).toMatchSnapshot();
  });
});
