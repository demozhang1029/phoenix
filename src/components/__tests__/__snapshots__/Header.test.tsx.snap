// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render header correctly 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "height": "100%",
        }
      }
      testID="header"
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 30,
          }
        }
      >
        hah
      </Text>
    </View>
  </View>
</View>
`;
